spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -388,14 +397,21 @@ When the @LRA annotation is defined multiple times, this is the order of precede
 More specifically, when the JAX-RS method and the class containing the JAX-RS method both have the
 `@LRA` annotation, the one from the JAX-RS method should be used.
 
-If the method is to run in the context of an LRA and the annotated class
-also contains a method annotated with `@Compensate` then the class
+If the method is to run in the context of an LRA then the annotated class
+MUST either contain a method annotated with `@Compensate` or `@AfterLRA`.
+If the `@Compensate` annotation is present then the class
 (aka participant or compensator) will be enlisted with the LRA.
-This means that before starting the method the implementation must be able
-to guarantee that participant has been or will eventually be registered
-with the LRA. The practical consequence of this requirement is that the
-implementation must durably record that the participant is enlisted before
-letting the business invocation proceed.
+If the `@AfterLRA` annotation is present then the class will be notified
+when the LRA reaches one of <<lra-state-model,the final LRA states>>.
+In the `@Compensate` case this means that before starting the method the implementation must
+be able to guarantee that the participant has been or will eventually be registered
+with the LRA. This implies that the LRA is effectively validated as being active (since
+it is not possible to enlist with an inactive LRA).
+In the `@AfterLRA` case the implementation must be able to guarantee that the
+listener will be notified when the LRA finishes. The practical consequence of this requirement
+is that the implementation must durably record that the participant/listener is associated
+with the LRA before letting the business invocation proceed.
```

Comment:
Maybe it would be worth to also mention that listener can be enlisted when the LRA is finishing (Cancelling or Closing).

Reply:
That kind of text should be in the javadoc or in the section "Discovering the Outcome of an LRA".
But I will add it with this PR since it relates to context validation.